fedimint-server/src/net/peers.rs

```diff
@@ -52,13 +52,15 @@ pub type PeerConnector<M> = AnyConnector<PeerMessage<M>>;
 /// [`FramedTransport`](crate::net::framed::FramedTransport) connections. For
 /// production deployments the `Connector` has to ensure that connections are
 /// authenticated and encrypted.
+#[derive(Clone)]
 pub struct ReconnectPeerConnections<T> {
     connections: HashMap<PeerId, PeerConnection<T>>,
 }
 
+#[derive(Clone)]
 struct PeerConnection<T> {
-    outgoing: Sender<T>,
-    incoming: Receiver<T>,
+    outgoing: async_channel::Sender<T>,
+    incoming: async_channel::Receiver<T>,
```

Comment:
I understand that this looks weird. We do this since aleph bft requires new a owned network layer struct for every session since its consumed by the run_session function and not returned. For the same reason we use and async channel to send proposals to the atomic broadcast. Though it looks weird its not an issue since only ever one clone of the struct is in use at a time as sessions are inherently sequential... I tried to find a different design but this seemed like the cleanest imo.

Reply:
Could completed sessions return the network object somehow? That way we can ensure that actually only one ever uses the network at the same time.